qt/widgets/instrumentview/src/Viewport.cpp

```diff
@@ -310,6 +322,33 @@ void Viewport::setViewToZNegative() {
   Mantid::Kernel::Quat tempy(180.0, Mantid::Kernel::V3D(0.0, 1.0, 0.0));
   m_quaternion = tempy;
   m_quaternion.GLMatrix(&m_rotationmatrix[0]);
+  adjustProjection(2);
+}
+
+void Viewport::adjustProjection(const unsigned int axis) {
+  // reset the projection bounds to the original values
+  m_left = m_leftOrig;
+  m_right = m_rightOrig;
+  m_top = m_topOrig;
+  m_bottom = m_bottomOrig;
+  m_near = m_nearOrig;
+  m_far = m_farOrig;
+  // rotate the original projection based on the new quaternion
+  m_quaternion.rotateBB(m_left, m_bottom, m_near, m_right, m_top, m_far);
+  if (axis == 0) {
+    // X axis rotation, so use rotated Z values as new X
+    m_left = m_near;
+    m_right = m_far;
```

Comment:
I don't think this is correct for all cases because `m_near` and `m_far` isn't set to `minBounds.Z(), maxBounds.Z()` in `Viewport::setProjection`?